lfs.c

```diff
@@ -4396,6 +4396,32 @@ static int lfs_format_(lfs_t *lfs, const struct lfs_config *cfg) {
 }
 #endif
 
+struct lfs_tortoise_t
+{
+    lfs_block_t pair[2];
+    lfs_size_t i;
+    lfs_size_t period;
+};
+
+static int lfs_detect_cycles(lfs_mdir_t dir, struct lfs_tortoise_t *tortoise)
```

Comment:
It looks like you're passing the `dir` by value, instead of by pointer. This creates a copy of the 8-word struct, which might be adding both stack cost and code cost.

Passing by pointer:
``` c
static int lfs_detect_cycles(const lfs_mdir_t *dir, struct lfs_tortoise_t *tortoise) {
```

Reply:
I missed it, it really shouldn't be, ^_^.
I've resubmitted it.